collectors/osv/collectors.py

```diff
@@ -0,0 +1,275 @@
+import json
+import re
+from io import BytesIO
+from typing import Union
+from zipfile import ZipFile
+
+import requests
+from celery.utils.log import get_task_logger
+from cvss import CVSS2, CVSS3
+from django.conf import settings
+from django.utils import timezone
+
+from collectors.constants import SNIPPET_CREATION_ENABLED
+from collectors.framework.models import Collector
+from osidb.core import set_user_acls
+from osidb.models import FlawCVSS, FlawReference, Snippet
+from osidb.validators import CVE_RE_STR
+
+logger = get_task_logger(__name__)
+
+
+class OSVCollector(Collector):
+    # Snippet creation is disabled for now
+    snippet_creation_enabled = None
+
+    SUPPORTED_OSV_ECOSYSTEMS = (
+        "Bitnami",  # General purpose Vulnerability Database: https://github.com/bitnami/vulndb
+        "Go",
+        # "Hackage",  # TODO: do we ship any Haskell packages?
+        # "Hex",  # TODO: do we ship any Erlang packages?
+        "Linux",  # Linux kernel vulnerabilities
+        "Maven",  # Java package ecosystem
+        "NuGet",  # .NET package ecosystem
+        "Packagist",  # Do we ship any PHP packages (besides PHP itself?)
+        # "Pub",  # TODO: Do we ship any Dart packages?
+        "PyPI",  # Python package ecosystem
+        "RubyGems",  # Ruby package ecosystem
+        "crates.io",  # Rust package ecosystem
+        "npm",  # JavaScript package ecosystem
+    )
+
+    # On top of fetching vulnerabilities for certain ecosystems only (see above), we also need to
+    # filter individual OSV vulnerabilities by their ID prefix. For example, we are not
+    # interested in reported malicious packages (in any ecosystem) that have a prefix of "MAL-".
+    OSV_VULN_ID_SKIPLIST = (
+        # Malicious packages as recorded in https://github.com/ossf/malicious-packages
+        "MAL",
+        # Global Security Database: https://github.com/cloudsecurityalliance/gsd-database
+        # Too noisy; every single kernel commit seems to get an auto-assigned GSD identifier
+        # without actually determining if it fixes a security issue.
+        "GSD",
+    )
+
+    # Browseable web UI: https://console.cloud.google.com/storage/browser/osv-vulnerabilities
+    OSV_DATA_SNAPSHOT_URL = "https://storage.googleapis.com/osv-vulnerabilities"
+    # API docs: https://google.github.io/osv.dev/api/
+    OSV_VULNS_API_URL = "https://api.osv.dev/v1/vulns"
+
+    def __init__(self):
+        """initiate collector"""
+        super().__init__()
+        if self.snippet_creation_enabled is None:
+            self.snippet_creation_enabled = SNIPPET_CREATION_ENABLED
+
+    def fetch_osv_vulns_for_ecosystem(self, ecosystem: str) -> dict:
+        # Download the ZIP file from the URL
+        url = f"{self.OSV_DATA_SNAPSHOT_URL}/{ecosystem}/all.zip"
```

Comment:
No, the API only provides the ability to query by package identifiers (like name or purl) and version identifiers (version or commit): https://google.github.io/osv.dev/post-v1-query/

I might submit an issue to request fetching by timestamp (modified) but I doubt it would be implemented before this PR is merged. For now, fetching everything seems fine. The OSV records are relatively small (unlike our VEX files for example which are very verbose) so the processing of each ecosystem's ZIP is not too resource intensive.